modelator/utils/modelator_helpers.py

```diff
@@ -2,6 +2,61 @@
 from typing import Dict
 from .. import const_values
 import logging
+import subprocess
+from urllib.request import urlopen
+import zipfile
+import io
+
+
+def check_for_apalache_jar(
+    download_location=const_values.DEFAULT_APALACHE_LOCATION,
+    jar_path=const_values.DEFAULT_APALACHE_JAR,
+    expected_version=const_values.DEFAULT_APALACHE_VERSION,
+) -> bool:
+
+    """
+    Checks for existence of the Apalache jar, version `expected_version`.
+    If it is missing, downloads it to `download_location` and returns False.
+    If it is already there, does nothing and returns True.
+
+    It is meant to be used with default arguments.
+    The reason that arguments exist is to enable testing.
+    """
+
+    apalache_release_url = "https://github.com/informalsystems/apalache/releases/download/v{}/apalache.zip".format(
+        expected_version
+    )
+
+    logging.debug("checking for jar at {}".format(jar_path))
+    try:
+        download_needed = False
+        version = subprocess.check_output(
+            ["java", "-jar", jar_path, "version"], text=True, stderr=subprocess.STDOUT
+        ).strip()
+        logging.debug(
+            "Currently existing version is {} and we are looking for {}".format(
+                version, expected_version
+            )
+        )
+        if not version == expected_version:
+            download_needed = True
+    except subprocess.CalledProcessError:
+        logging.debug("Error checking version of the jar")
+        download_needed = True
+    finally:
+        if download_needed is True:
+            with urlopen(apalache_release_url) as zip_response:
+                with zipfile.ZipFile(io.BytesIO(zip_response.read())) as zip_file:
```

Comment:
```
import hashlib
data = zip_response.read()
assert APALACHE_SHA256 == hashlib.sha256(data).hexdigest()
with zipfile.ZipFile(io.BytesIO(data)) as zip_file:
```

Reply:
in that case, whenever bumping a version, it is the responsibility of the user to update the SHA as well?

Reply:
In case that there was some network error while downloading, our current logic would re-download. 
A malicious attack on Apalache repository? That seems to be equally likely as a malicious attack on our repository (and, thus, tampering with the SHA value).

Reply:
1. it's us who bump the version (and the checksum). you hardcoded the version in the code.
2. Are you saying we don't need checksum verification?

Reply:
2. Yes, I am aiming at that.

Reply:
As discussed in person: we still need a checksum because when Modelator will be packaged and its own data checked against the checksum, pulling Apalache jar without checking would be the weakest link in the chain.